applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -270,21 +270,21 @@ static int headset_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
 
 	if (IS_ENABLED(CONFIG_BT_AUDIO_PRES_DELAY_SRCH_PREF_MIN)) {
 		/* Preferred min is 0, so we set min supported */
-		if (pref_dly_min == 0) {
-			*pres_dly_us = pres_dly_min;
-		} else {
+		if (IN_RANGE(pref_dly_min, pres_dly_min, pres_dly_max)) {
 			*pres_dly_us = pref_dly_min;
+		} else {
+			*pres_dly_us = pres_dly_min;
 		}
```

Comment:
This code will set pres_dly_min even if pref_dly_min is larger than pres_dly_max.  It is a strange scenario, but I guess it could happen. We should rail the pref_min/pref_max at the start of the function to be within range of pres_min/pres_max and then continue with the rest of the code.

Reply:
Good spot!

I have updated the code to now test if the pref min or max is outside the pres min and max. It will also error if the pref_dly_max is less than the pres_dly_min or the pref_dly_min is greater than the pres_dly_max. If either of thess occur then there has been an error in the serach or defined delays.